markdown/developers/pipeline_structure.md

```diff
@@ -0,0 +1,74 @@
+# nf-core contribution guidelines
+
+Nf-core pipelines adhere to common design principles that have been crafted through discussions and extensive real-world experiences with pipelines over time. We understand, however, that not all about nf-core pipelines is obvious and can be confusing to people new to nf-core. To help you along getting started with nf-core, writing your pipeline or contribute otherwise, we have created this documentation. Hopefully you will find most things that might confuse you described here, and if not, be sure to bug the nf-core Slack channel!
+
+## General
+
+The best way to write a new pipeline is to start from the template (see here). After doing that, you will have a number of files in your repository with different purposes. Here's what they are for:
+
+### Files
+
+* `main.nf`: This is the main nextflow file which will get executed if the pipeline is run. It parses all parameters, loads the modules with options and defines the specific order in which these are executed.
+
+* `nextflow.config`: The main nextflow configuration file. It contains the default pipeline parameters, nextflow configuration options and information like pipeline and nextflow version, among others.
+
+* `README.md`: Basic information about the pipeline and usage
+
+* `nextflow_json.schema`: The JSON schema file is used for pipeline parameter specification. It can be automatically created usint the `tools` package. In the pipeline, this is used for printing out a help message to the user or validating the input parameters. The nf-core tools utility package furthermore uses the JSON Schema for launching pipelines. Furthermore, `nextflow.config` defines different profiles that can be used to run the pipeline. These include the test profiles, which load, when used, the above mentioned configuration files into the `nextflow.config` file. Additionaly, the following profiles are available by default: `debug`, `conda`, `docker`, `singularity` and `podman`‚ which are used to run the pipeline in debug mode, using a conda environment or a using a container technology (docker, singularity or podman).
```

Comment:
I didn't really understand the bit about loading stuff into the config file? Anyway I moved the bit about `nextflow.config` into the bullet for that file and rewrote a little to point to the configs docs instead.

```suggestion
* `nextflow_json.schema`: The JSON schema file is used for pipeline parameter specification. This is automatically created using the `nf-core schema build` command. It is used for printing command-line help, validating input parameters, building the website docs and for building pipeline launch interfaces (web and cli).
```